### PR TITLE
merge activity settings for calendars, events and todos

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Fine with me

**Comment by jancborchardt** [comment]
Please review @nextcloud/calendar @nextcloud/tasks :)

**Comment by georgehrke** [comment]
What is the intention here? Is there any related Github issue?

**Comment by rullzer** [comment]
To have cleaner notifiation settings

**Comment by MorrisJobke** [comment]
Before:
Bildschirmfoto 2020-08-11 um 21 53 30
After:
Bildschirmfoto 2020-08-11 um 21 53 18

**Comment by rullzer** [comment]
> Before:
> Bildschirmfoto 2020-08-11 um 21 53 30
> After:
> Bildschirmfoto 2020-08-11 um 21 53 18

I think you switched before and after ;)

**Review by rullzer — APPROVED** [review]

**Review by georgehrke — COMMENTED** [review]
Code-wise the change is fine.

Not really happy about it though.

**Comment by tcitworld** [comment]
Yeah, don't really like it either, pretty sure people won't be happy not having separate settings anymore. For instance, the deletion of a (shared) calendar is a pretty much important event that people may want to subscribe to without knowing of every event or task updates.

Why not use #22117 to create a proper dav group if the issue is too many checkboxes?

**Comment by rullzer** [comment]
cc: @jancborchardt ^^^

**Comment by jancborchardt** [comment]
To have the settings cleaner but also useful, we should not have the Calendar and Tasks settings in "Other activities" but rather a specific section "Calendar and Tasks" – I assume that’s what you meant @tcitworld, right?

In there we can then have entries for:

- A calendar was modified
- A calendar was deleted
- A calendar event was modified
- A calendar event was deleted
- A task was modified
- A task was deleted

Would that work @georgehrke @tcitworld?

**Comment by tcitworld** [comment]
Right now we don't differentiate between calendar/event/tasks modifications or deletions, but yeah, that seems about right.

File: apps/dav/appinfo/info.xml [hunk 1]
```diff
@@ -60,13 +60,10 @@
 	<activity>
 		<settings>
 			<setting>OCA\DAV\CalDAV\Activity\Setting\Calendar</setting>
-			<setting>OCA\DAV\CalDAV\Activity\Setting\Event</setting>
-			<setting>OCA\DAV\CalDAV\Activity\Setting\Todo</setting>
 		</settings>
 
 		<filters>
 			<filter>OCA\DAV\CalDAV\Activity\Filter\Calendar</filter>
-			<filter>OCA\DAV\CalDAV\Activity\Filter\Todo</filter>
```
Comment (nickvergessen) [inline apps/dav/appinfo/info.xml:69]: Merging the settings fine by me, but the filter is really useful I think, as when you use fully interact with todos, the calendar changes are hard to find if they are all in one filter.

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by nickvergessen** [comment]
I think the list could be simplified, but this is the minimum:


- A calendar was modified (created, deleted, shared, ...)
- A calendar event was modified (created, deleted, updated same as files)
- A task was modified (created, deleted, updated same as files)

**Comment by jancborchardt** [comment]
> I think the list could be simplified, but this is the minimum:
> * A calendar was modified (created, deleted, shared, ...)
> * A calendar event was modified (created, deleted, updated same as files)
> * A task was modified (created, deleted, updated same as files)

Then let’s do that and put it in a section "Calendar and tasks" so it’s not just in "Other". @icewind1991 can you do that?

**Comment by nickvergessen** [comment]
No, he is on vacation

**Comment by icewind1991** [comment]
Changed:

image

File: apps/dav/lib/CalDAV/Activity/Setting/CalDAVSetting.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	public function __construct(IL10N $l) {
+		$this->l = $l;
+	}
+
+	public function getGroupIdentifier() {
+		return 'calendar';
+	}
+
+	public function getGroupName() {
+		return $this->l->t("Calendar and tasks");
```
Comment (nickvergessen) [inline apps/dav/lib/CalDAV/Activity/Setting/CalDAVSetting.php:45]:
```suggestion
		return $this->l->t('Calendar and tasks');
```

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**